### PR TITLE
Refactor/pn 12178 - Translate common components using translation hook

### PR DESCRIPTION
**Pull request by mflauti**
This PR refactors the code for the common components (header, menu, footer) of the page structure making use of the translation hook to support multilanguage functionality

### CODE REVIEW

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: src/api/data/it.tsx [hunk 1]
```diff
@@ -13,15 +13,15 @@ import {
...
-import {
-  assistanceLink,
-  companyLegalInfo,
-  navigation,
-  pagoPALink,
-  postLoginLinks,
-  preLoginLinks,
-  productJson,
-} from "./it/common";
+// import {
```
[the remark this answers is not in the document]
Comment (mflauti) [reply src/api/data/it.tsx:16]: ok

File: public/locales/en/common.json [hunk 2]
```diff
@@ -1,9 +1,146 @@
...
+        "ariaLabel": "Link: vai al sito di PagoPA S.p.A."
+    },
     "footer": {
         "it": "Italian",
         "en": "English",
         "fr": "French",
         "de": "German",
-        "sl": "Slovenian"
+        "sl": "Slovenian",
+        "company_info": {
```
[the remark this answers is not in the document]
Comment (mflauti) [reply public/locales/en/common.json:12]: ok

File: src/components/Layout.tsx [hunk 3]
```diff
@@ -85,24 +84,25 @@ const Layout: React.FC<Props> = ({ children }) => {
             </ButtonNaked> */}
           </Stack>
         </Stack>
-        <NavigationBar {...appData.common.navigation} />
+        <NavigationBar title={t('navigation.title')} image={`${IMAGES_PATH}/logo.svg`} />
```
[the remark this answers is not in the document]
Comment (mflauti) [reply src/components/Layout.tsx:87]: ok

File: src/components/Layout.tsx [hunk 4]
```diff
@@ -85,24 +84,25 @@ const Layout: React.FC<Props> = ({ children }) => {
...
             onClick: () =>
-              window.open(appData.common.pagoPALink.href, "_blank"),
+              window.open(PAGOPA_HOME, "_blank"),
           }}
-          legalInfo={appData.common.companyLegalInfo}
-          postLoginLinks={appData.common.postLoginLinks}
-          preLoginLinks={appData.common.preLoginLinks(windowURL)}
+          legalInfo={companyLegalInfo(t, lang)}
+          postLoginLinks={postLoginLinks(t, lang)}
+          // preLoginLinks={preLoginLinks(t, lang)}
```
[the remark this answers is not in the document]
Comment (mflauti) [reply src/components/Layout.tsx:100]: ok

File: src/components/NavigationBar.tsx [hunk 5]
```diff
@@ -100,7 +76,33 @@ const NavigationBar: React.FC<INavigationBarProps> = ({ title, image }) => {
...
+      path: "/pubbliche-amministrazioni",
+      subMenu: [
+        {
+          label: t('navigation.documentation'),
+          path: "/pubbliche-amministrazioni/documenti",
+        },
+      ],
+    },
+    
+    // {
```
[the remark this answers is not in the document]
Comment (mflauti) [reply src/components/NavigationBar.tsx:95]: ok

File: src/pages/[lang]/punti-di-ritiro/index.dev.tsx [hunk 6]
```diff
@@ -245,7 +264,12 @@ const RitiroPage: NextPage = () => {
         </Box>
       )}
 
-      <DarkInfoblockRitiro></DarkInfoblockRitiro>
+      <DarkInfoblockRitiro
```
[the remark this answers is not in the document]
Comment (mflauti) [reply src/pages/[lang]/punti-di-ritiro/index.dev.tsx:267]: ok

**Review by AndreaCimini90 — APPROVED** [review]

File: public/locales/it/common.json [hunk 7]
```diff
@@ -1,9 +1,152 @@
...
+            "login": "Accedi"
+        },
+        "enti": {
+            "title": "Sei un ente?",
+            "description": "Il tuo ente non ha ancora aderito?",
+            "link": "Scopri come aderire",
+            "login": "Accedi"
+        }
+    },
+    "pickup_points": {
```
Comment (AndreaCimini90) [inline public/locales/it/common.json:140]: move pickup_points and infoblock_ritiro into a file named ritiro.json

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**